R/github-actions.R

```diff
@@ -70,13 +70,18 @@ use_github_action <- function(name = NULL,
                               open = FALSE,
                               badge = NULL) {
 
-  if (is.null(url) && is.null(name)) {
-    name <- choose_gha_workflow()
-  }
+  maybe_name(name)
+  maybe_name(ref)
+  maybe_name(url)
+  maybe_name(save_as)
+  maybe_name(readme)
+  check_bool(ignore)
+  check_bool(open)
+  check_bool(badge, allow_null = TRUE)
```

Comment:
I haven't done an analysis, but assume you have. Moving these all up here and as a result, softening some `check_`s to `maybe_`s must change behaviour somewhat. But presumably not in any case we're actually worried about.

Reply:
I've looked at this one pretty carefully and I'm fairly certain it doesn't meaningfully change the behaviour - In most cases it just consolidates the logic:

- `name` and `ref` are only used when `url` is `NULL` and the case when `name` is `NULL` is caught on line 84, and `ref` is `NULL` on line 90. Otherwise `maybe_name()` ensures they are length 1 character vectors.
- `maybe_name(url)` combined with `if (is.null(url))` on line 82 ensures that if `url` is not `NULL` it will be a length 1 character vector.
- The cases where `save_as` and `readme` being null are already caught on lines 100 and 120 respectively.